code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,454 @@
+    }
+
+    constructor(extensionApi: ExtensionApi) : this(extensionApi, ServiceProvider.getInstance())
+
+    /**
+     * Exists only for cascading components for dependency injection.
+     */
+    private constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider
+    ) : this(
+        extensionApi, serviceProvider,
+        AppIdManager(serviceProvider.dataStoreService, serviceProvider.deviceInfoService),
+        CacheManager(serviceProvider.deviceInfoService),
+        LaunchRulesEvaluator("Configuration", LaunchRulesEngine(extensionApi), extensionApi),
+        Executors.newSingleThreadScheduledExecutor()
+    )
+
+    /**
+     * Exists only for cascading components for dependency injection.
+     */
+    private constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider,
+        appIdManager: AppIdManager,
+        cacheFileService: CacheFileService,
+        launchRulesEvaluator: LaunchRulesEvaluator,
+        retryWorker: ScheduledExecutorService
+    ) : this(
+        extensionApi,
+        serviceProvider,
+        appIdManager,
+        cacheFileService,
+        launchRulesEvaluator,
+        retryWorker,
+        ConfigurationStateManager(
+            appIdManager,
+            cacheFileService,
+            serviceProvider.networkService,
+            serviceProvider.deviceInfoService,
+            serviceProvider.dataStoreService
+        ),
+        ConfigurationRulesManager(
+            launchRulesEvaluator,
+            serviceProvider.dataStoreService,
+            serviceProvider.networkService,
+            cacheFileService
+        )
+    )
+
+    @VisibleForTesting
+    internal constructor(
+        extensionApi: ExtensionApi,
+        serviceProvider: ServiceProvider,
+        appIdManager: AppIdManager,
+        cacheFileService: CacheFileService,
+        launchRulesEvaluator: LaunchRulesEvaluator,
+        retryWorker: ScheduledExecutorService,
+        configurationStateManager: ConfigurationStateManager,
+        configurationRulesManager: ConfigurationRulesManager
+    ) : super(extensionApi) {
+        this.serviceProvider = serviceProvider
+        this.appIdManager = appIdManager
+        this.cacheFileService = cacheFileService
+        this.launchRulesEvaluator = launchRulesEvaluator
+        this.retryWorker = retryWorker
+        this.configurationStateManager = configurationStateManager
+        this.configurationRulesManager = configurationRulesManager
+    }
+
+    /**
+     * Sets up the Configuration extension for the SDK by loading the appId, publishing initial state
+     * and registering relevant event listener for processing events.
+     */
+    override fun onRegistered() {
+        super.onRegistered()
+
+        // TODO: Register pre-processor
+        api.registerEventListener(
+            EVENT_TYPE,
+            EVENT_SOURCE_REQUEST_CONTENT,
+            extensionEventListener
+        )
+
```

Comment:
https://github.com/adobe/aepsdk-core-android/blob/791777fcddf803d7b764d37e125732b42cac706f/code/android-core-library/src/main/java/com/adobe/marketing/mobile/ConfigurationExtension.java#L246
We did loading local (bundled/cached) config when the boot event is dispatched, why do we change this logic now? Btw, I think extensions should only do `registerEventListener` operations when `onRegistered` is called.

Reply:
The new EventHub, as of now, does not send a boot event when start is called. As the extensions are processing their own queue via the container, I felt that it should be sufficient to move the operations done during boot earlier to onRegistered, similar to the [implementation swift](https://github.com/adobe/aepsdk-core-ios/blob/main/AEPCore/Sources/configuration/Configuration.swift#L44).  Do you see any down side of doing it this way?